### ISSUE TITLE
Investigate an initContainer style approach rather than outer/inner containers

### ISSUE THREAD

**Issue by binocarlos** [dotmesh-io/dotmesh#327]
`dm cluster init` runs [require_zfs.sh](https://github.com/dotmesh-io/dotmesh/blob/master/cmd/dotmesh-server/require_zfs.sh) which initialises the shared mount points on the host before starting an inner container that uses `-v $MOUNTPOINT:$MOUNTPOINT:rshared` consuming the host mount point.

It should be possible for `dm cluster init` to first run an `initContainer` in the style of k8s which does the host mount point setup and then exists before starting a second container that consumes the mountpoint using `:rshared`

In essence - the outer container exists in order to do some initialisation - this issue involves investigating if we can run the initialisation container then the actual dotmesh comntainer **sequentially** rather than one wrapping the other.

If this works - and once GKE clusters supports shared mount points [not in alpha](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) - it means we can run both Kubernetes and normal `dm cluster init` clusters both in the same way.

This issue involves investigating if this is possible and what other issues might surface from doing it this way.

**Comment by binocarlos** [comment]
OK having talked to @alaric-datamesh about this - a problem would be that if the host restarts - `modprobe zfs` and `mount --make-shared $MOUNTPOINT;` would need to be re-run.  This would not work if Docker simply restarts the dotmesh-server that expects these commands to already have happened.

We need a way of the single container both initializing and co-ordinating the start of the actual server container without the container names clashing.

**Comment by alaric-dotmesh** [comment]
Duplicate of #275
